packages/loader/container-loader/src/deltaManager.ts

```diff
@@ -1186,9 +1192,14 @@ export class DeltaManager
 
         // if we have connection, and message is local, then we better treat is as local!
         assert(!this.connection || this.connection.details.clientId !== message.clientId ||
-            this.lastSubmittedClientId === message.clientId, "Not accounting local messages correctly");
+            this.lastSubmittedClientIds.includes(message.clientId), "Not accounting local messages correctly");
 
-        if (this.lastSubmittedClientId !== undefined && this.lastSubmittedClientId === message.clientId) {
+        const foundIndex = this.lastSubmittedClientIds.indexOf(message.clientId);
+        if (foundIndex >= 0) {
+            // Report if we're getting messages from a clientId older than the most recent
+            if (foundIndex !== this.lastSubmittedClientIds.length - 1) {
```

Comment:
>lastSubmittedClientIds [](start = 36, length = 22)

i don't think this would catch the issue. the new client id never sent any messages  until after it's join and after the messages from the pervious client

Reply:
i think you'll actually have to make a change like this in Container. 

---
In reply to: [487345319](https://github.com/microsoft/FluidFramework/pull/3613#discussion_r487345319) [](ancestors = 487345319)

Reply:
the check should happen in processRemoteMessage, and should push client ids in initializeProtocolState in the addMember callback after we set the connection state

---
In reply to: [487345837](https://github.com/microsoft/FluidFramework/pull/3613#discussion_r487345837) [](ancestors = 487345837,487345319)